cli/config/configfile/file.go

```diff
@@ -152,6 +163,39 @@ func (configFile *ConfigFile) Save() error {
 	return configFile.SaveToWriter(f)
 }
 
+// ParseProxyConfig computes proxy configuration by retreiving the config for the provided host and
+// then checking this against any environment variables provided to the container
+func (configFile *ConfigFile) ParseProxyConfig(host string, runOpts opts.ListOpts) map[string]*string {
```

Comment:
This function only needs a `[]string` since it's calling `GetAll()` on `runOpts` and not using it in any other way. Could we have it accept a `[]string` instead? That should make testing easier as well.